### PR TITLE
Added new option in Matomo Config Variable to disable Form and Media Analytics

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Added new option in Matomo Config Variable to disable Form and Media Analytics.
Fixes: #PG-657

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by snake14 — CHANGES_REQUESTED** [review]
Looks good and the functional testing went well once I remembered that I had to publish a new version after updating my configuration variable. However, I noticed that I had another configuration variable showing as changed when I hadn't changed it. I think that you need to create a migration and do something similar to `plugins/TagManager/Updates/NewTagParameterMigrator.php` in order to add the new params to all existing config variables.

**Comment by AltamashShaikh** [comment]
> Looks good and the functional testing went well once I remembered that I had to publish a new version after updating my configuration variable. However, I noticed that I had another configuration variable showing as changed when I hadn't changed it. I think that you need to create a migration and do something similar to `plugins/TagManager/Updates/NewTagParameterMigrator.php` in order to add the new params to all existing config variables.

@snake14 Added migration script,
To test it rename the file to `4.12.0-b1` and your version in the `option` table to `4.11.0` and give it a try, thats how I was able to test it

File: Updates/NewVariableParameterMigrator.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        }
+
+        foreach ($activeVariableIds as $idVariable) {
+            $this->updateVariableParameters($idSite, $idVersion, $idVariable);
+        }
+    }
+
+    private function updateVariableParameters($idSite, $idVersion, $idVariable)
+    {
+        $variableInfo = $this->variablesDao->getContainerVariable($idSite, $idVersion, $idVariable);
```
Comment (snake14) [inline Updates/NewVariableParameterMigrator.php:100]: I had to add the following below this line in order to get it to run for all of my sites:
```
if (empty($variableInfo['parameters'])) {
        return;
}
```

Once I added that check, it updated the necessary variables and they no longer showed as changed when I went to publish a new version.

**Review by snake14 — CHANGES_REQUESTED** [review]
Looks good. I had to make one little change to get the functional test to work. Please see my inline comment.

File: Updates/NewVariableParameterMigrator.php (hunk 1 above)
Comment (AltamashShaikh) [reply Updates/NewVariableParameterMigrator.php:100]: @snake14 updated as suggested

**Review by snake14 — APPROVED** [review]
Looks good :+1:

**Pull request merged**